cmd/main.go

```diff
@@ -54,20 +79,21 @@ func handleFlags(args []string) *rpc.Flags {
 			FullTimestamp: true,
 		})
 	}
-	return flags
+	return flags, true
 }
 
 func main() {
-	// ensure we are admin/sudo
-	checkAdminAccess()
-	runRPC(os.Args)
-}
-
-func checkAdminAccess() {
-	amt := amt.NewAMTCommand()
-	result, err := amt.Initialize()
-	if !result || err != nil {
-		log.Error(err) //Print the errors
-		os.Exit(1)
+	status, err := checkAccess()
+	if status != utils.Success {
+		if err != nil {
+			log.Error(err.Error())
+		}
+		log.Error(AccessErrMsg)
+		return
+	}
+	status, err = runRPC(os.Args)
+	if err != nil {
+		log.Error(err.Error())
 	}
+	log.Info("Completed with status code: ", status)
```

Comment:
I think this needs to remain os.Exit(status) -- this call must happen otherwise 0 is always the return code regardless of what we log it to be. It is only the runRPC function that is used by the library so what happens in main here can remain as os.Exit. I think we also should remove the log statement it doesn't tell us anything different than what the OS would tell me if you use os.exit as we should. If you want to keep it, put it as trace level. But if you do decide to keep it, you'll need to inspect the code and accurately log an error if it is not 0 or trace if it is 0

Reply:
ok, makes sense, I'll fix it.